### PR TITLE
feat(search): implement customizable search trigger modes with button enter, and instant options

### PR DESCRIPTION
**Pull request by balazs-szucs**
## 📝 Description

Closes: #3010

This pull request adds a user-configurable search trigger mode for both the book browser and header search bar, allowing users to choose how searches are initiated: instantly as they type, on pressing Enter, or via a dedicated button. The search mode preference is persisted and can be managed through a new settings panel. The implementation involves updates to the UI, logic for handling each mode, a new service for managing preferences, and comprehensive tests.

**Linked Issue:** Fixes #

> **Required.** Every PR must reference an approved issue. If no issue exists, [open one](https://github.com/booklore-app/booklore/issues/new) and wait for maintainer approval before submitting a PR. Unsolicited PRs without a linked issue will be closed.

## 🏷️ Type of Change

- [ ] Bug fix
- [ ] New feature
- [X] Enhancement to existing feature
- [ ] Refactor (no behavior change)
- [ ] Breaking change (existing functionality affected)
- [ ] Documentation update

## 🔧 Changes

The most important changes are:

**User Preference Management**
* Added a new `SearchPreferenceService` to manage and persist the user's search trigger mode (`instant`, `enter`, or `button`) using local storage.
* Integrated the new search trigger mode preference into both `BookBrowserComponent` and `BookSearcherComponent`, updating their logic to respect the selected mode for search initiation. 

**UI Enhancements**
* Added a new settings panel (`SearchPreferencesComponent`) in the view preferences section, allowing users to select their preferred search trigger mode with descriptive UI and instant feedback.
* Updated the book browser and header search bar templates to show the search button only when the 'button' mode is selected and to handle Enter key events appropriately.

**Testing**
* Added comprehensive unit tests for the search trigger logic, verifying correct behavior for all modes and ensuring robust preference handling.

**Internationalization**
* Extended English and German i18n files with new strings for the search preferences UI and feedback messages.

## 🧪 Testing (MANDATORY)

> **PRs without this section filled out will be closed.** "Tests pass" or "Tested locally" is not sufficient. You must provide specifics.

**Manual testing steps you performed:**

1. tested each search function in book view
2. confirm when button/enter mode is active then no instant search happening
3. confirmed button and enter works intended

**Regression testing:**

-

**Edge cases covered:**

-

**Test output:**

<details>
<summary>Backend test output (<code>./gradlew test</code>)</summary>

```
bszuc@brios:~/booklore/booklore-api$ ./gradlew clean test
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by lombok.permit.Permit
WARNING: Please consider reporting this to the maintainers of class lombok.permit.Permit
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release

> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Management of bidirectional association persistent attributes is deprecated and will be removed. Set the value to 'false' to get rid of this warning

> Task :compileTestJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by net.bytebuddy.dynamic.loading.ClassInjector$UsingUnsafe$Dispatcher$CreationAction (file:/home/bszuc/.gradle/caches/modules-2/files-2.1/net.bytebuddy/byte-buddy/1.17.8/af5735f63d00ca47a9375fae5c7471a36331c6ed/byte-buddy-1.17.8.jar)
WARNING: Please consider reporting this to the maintainers of class net.bytebuddy.dynamic.loading.ClassInjector$UsingUnsafe$Dispatcher$CreationAction
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release
2026-02-26T14:41:24.439+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.s.m.s.b.SimpleBrokerMessageHandler     : Stopping...
2026-02-26T14:41:24.440+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.s.m.s.b.SimpleBrokerMessageHandler     : BrokerAvailabilityEvent[available=false, SimpleBrokerMessageHandler [org.springframework.messaging.simp.broker.DefaultSubscriptionRegistry@257b0c96]]
2026-02-26T14:41:24.440+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.s.m.s.b.SimpleBrokerMessageHandler     : Stopped.
2026-02-26T14:41:24.453+01:00  INFO 290736 --- [booklore-api] [opFolderWatcher] o.b.s.b.BookdropMonitoringService        : Bookdrop monitor thread interrupted
2026-02-26T14:41:24.453+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.b.BookdropMonitoringService        : Stopped bookdrop folder monitor
2026-02-26T14:41:24.839+01:00  INFO 290736 --- [booklore-api] [opFileProcessor] o.b.s.b.BookdropEventHandlerService      : File processing thread interrupted, shutting down.
2026-02-26T14:41:24.840+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.monitoring.MonitoringService       : Shutting down monitoring service...
2026-02-26T14:41:24.840+01:00  WARN 290736 --- [booklore-api] [        async-0] o.b.service.monitoring.MonitoringTask    : WatchService has been closed. Stopping monitoring.
2026-02-26T14:41:24.840+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.watcher.LibraryFileEventProcessor  : Shutting down LibraryFileEventProcessor...
2026-02-26T14:41:24.846+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2026-02-26T14:41:24.867+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown initiated...
2026-02-26T14:41:24.868+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown completed.
2026-02-26T14:41:24.873+01:00  INFO 290736 --- [booklore-api] [opFolderWatcher] o.b.s.b.BookdropMonitoringService        : Bookdrop monitor thread interrupted
2026-02-26T14:41:24.876+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.b.BookdropMonitoringService        : Stopped bookdrop folder monitor
2026-02-26T14:41:24.876+01:00  INFO 290736 --- [booklore-api] [opFileProcessor] o.b.s.b.BookdropEventHandlerService      : File processing thread interrupted, shutting down.
2026-02-26T14:41:24.877+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.monitoring.MonitoringService       : Shutting down monitoring service...
2026-02-26T14:41:24.878+01:00  WARN 290736 --- [booklore-api] [        async-0] o.b.service.monitoring.MonitoringTask    : WatchService has been closed. Stopping monitoring.
2026-02-26T14:41:24.878+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.watcher.LibraryFileEventProcessor  : Shutting down LibraryFileEventProcessor...
2026-02-26T14:41:24.886+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2026-02-26T14:41:24.889+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown initiated...
2026-02-26T14:41:24.889+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown completed.
2026-02-26T14:41:24.894+01:00  INFO 290736 --- [booklore-api] [opFolderWatcher] o.b.s.b.BookdropMonitoringService        : Bookdrop monitor thread interrupted
2026-02-26T14:41:24.894+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.b.BookdropMonitoringService        : Stopped bookdrop folder monitor
2026-02-26T14:41:24.895+01:00  INFO 290736 --- [booklore-api] [opFileProcessor] o.b.s.b.BookdropEventHandlerService      : File processing thread interrupted, shutting down.
2026-02-26T14:41:24.896+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.monitoring.MonitoringService       : Shutting down monitoring service...
2026-02-26T14:41:24.896+01:00  WARN 290736 --- [booklore-api] [        async-0] o.b.service.monitoring.MonitoringTask    : WatchService has been closed. Stopping monitoring.
2026-02-26T14:41:24.896+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.watcher.LibraryFileEventProcessor  : Shutting down LibraryFileEventProcessor...
2026-02-26T14:41:24.903+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2026-02-26T14:41:24.912+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown initiated...
2026-02-26T14:41:24.913+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown completed.
2026-02-26T14:41:24.919+01:00  INFO 290736 --- [booklore-api] [opFolderWatcher] o.b.s.b.BookdropMonitoringService        : Bookdrop monitor thread interrupted
2026-02-26T14:41:24.919+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.b.BookdropMonitoringService        : Stopped bookdrop folder monitor
2026-02-26T14:41:24.919+01:00  INFO 290736 --- [booklore-api] [opFileProcessor] o.b.s.b.BookdropEventHandlerService      : File processing thread interrupted, shutting down.
2026-02-26T14:41:24.920+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.monitoring.MonitoringService       : Shutting down monitoring service...
2026-02-26T14:41:24.921+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.watcher.LibraryFileEventProcessor  : Shutting down LibraryFileEventProcessor...
2026-02-26T14:41:24.921+01:00  WARN 290736 --- [booklore-api] [        async-0] o.b.service.monitoring.MonitoringTask    : WatchService has been closed. Stopping monitoring.
2026-02-26T14:41:24.927+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2026-02-26T14:41:24.937+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown initiated...
2026-02-26T14:41:24.938+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown completed.
2026-02-26T14:41:24.944+01:00  INFO 290736 --- [booklore-api] [opFolderWatcher] o.b.s.b.BookdropMonitoringService        : Bookdrop monitor thread interrupted
2026-02-26T14:41:24.944+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.b.BookdropMonitoringService        : Stopped bookdrop folder monitor
2026-02-26T14:41:24.945+01:00  INFO 290736 --- [booklore-api] [opFileProcessor] o.b.s.b.BookdropEventHandlerService      : File processing thread interrupted, shutting down.
2026-02-26T14:41:24.946+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.monitoring.MonitoringService       : Shutting down monitoring service...
2026-02-26T14:41:24.946+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.watcher.LibraryFileEventProcessor  : Shutting down LibraryFileEventProcessor...
2026-02-26T14:41:24.946+01:00  WARN 290736 --- [booklore-api] [        async-0] o.b.service.monitoring.MonitoringTask    : WatchService has been closed. Stopping monitoring.
2026-02-26T14:41:24.952+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2026-02-26T14:41:24.963+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown initiated...
2026-02-26T14:41:24.964+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown completed.
2026-02-26T14:41:24.967+01:00  INFO 290736 --- [booklore-api] [opFolderWatcher] o.b.s.b.BookdropMonitoringService        : Bookdrop monitor thread interrupted
2026-02-26T14:41:24.967+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.b.BookdropMonitoringService        : Stopped bookdrop folder monitor
2026-02-26T14:41:24.967+01:00  INFO 290736 --- [booklore-api] [opFileProcessor] o.b.s.b.BookdropEventHandlerService      : File processing thread interrupted, shutting down.
2026-02-26T14:41:24.968+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.monitoring.MonitoringService       : Shutting down monitoring service...
2026-02-26T14:41:24.968+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] o.b.s.watcher.LibraryFileEventProcessor  : Shutting down LibraryFileEventProcessor...
2026-02-26T14:41:24.968+01:00  WARN 290736 --- [booklore-api] [        async-0] o.b.service.monitoring.MonitoringTask    : WatchService has been closed. Stopping monitoring.
2026-02-26T14:41:24.971+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] j.LocalContainerEntityManagerFactoryBean : Closing JPA EntityManagerFactory for persistence unit 'default'
2026-02-26T14:41:24.981+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown initiated...
2026-02-26T14:41:24.982+01:00  INFO 290736 --- [booklore-api] [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : BookloreHikariPool - Shutdown completed.

[Incubating] Problems report is available at: file:///home/bszuc/booklore/booklore-api/build/reports/problems/problems-report.html

BUILD SUCCESSFUL in 1m 55s
7 actionable tasks: 7 executed
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.3.1/userguide/configuration_cache_enabling.html
```

</details>

<details>
<summary>Frontend test output (<code>ng test</code>)</summary>

```
bszuc@brios:~/booklore/booklore-ui$ npm run test

> booklore@0.0.0 test
> ng test

Using Vitest configuration file: /home/bszuc/booklore/booklore-ui/vitest-base.config.ts
Initial chunk files                                                            | Names                                                                       |  Raw size
spec-app-features-magic-shelf-component-magic-shelf-component.js               | spec-app-features-magic-shelf-component-magic-shelf-component               |   1.09 MB | 
spec-app-features-magic-shelf-service-book-rule-evaluator.service.js           | spec-app-features-magic-shelf-service-book-rule-evaluator.service           |  65.08 kB | 
spec-app-app.component.js                                                      | spec-app-app.component                                                      |  58.63 kB | 
chunk-AD2BXOLH.js                                                              | -                                                                           |  53.30 kB | 
styles.css                                                                     | styles                                                                      |  30.99 kB | 
chunk-NM2JNTGJ.js                                                              | -                                                                           |  24.50 kB | 
spec-app-features-magic-shelf-service-book-rule-evaluator-metadata-presence.js | spec-app-features-magic-shelf-service-book-rule-evaluator-metadata-presence |  23.21 kB | 
chunk-ZRYX4BS5.js                                                              | -                                                                           |  15.97 kB | 
spec-app-features-magic-shelf-service-magic-shelf-utils.js                     | spec-app-features-magic-shelf-service-magic-shelf-utils                     |  11.36 kB | 
spec-app-core-security-auth-initializer.js                                     | spec-app-core-security-auth-initializer                                     |   8.69 kB | 
spec-app-features-book-components-book-searcher-book-searcher.component.js     | spec-app-features-book-components-book-searcher-book-searcher.component     |   7.72 kB | 
chunk-PUIDBMXW.js                                                              | -                                                                           |   6.10 kB | 
chunk-Y7BNZHZM.js                                                              | -                                                                           |   4.86 kB | 
chunk-3J3QIHZF.js                                                              | -                                                                           |   2.19 kB | 
chunk-HPNCK62B.js                                                              | -                                                                           |   1.33 kB | 
init-testbed.js                                                                | init-testbed                                                                |   1.27 kB | 
spec-app-app.js                                                                | spec-app-app                                                                | 202 bytes | 
polyfills.js                                                                   | polyfills                                                                   | 121 bytes | 

                                                                               | Initial total                                                               |   1.41 MB

Application bundle generation complete. [7.625 seconds] - 2026-02-26T13:39:34.960Z

Watch mode enabled. Watching for file changes...

 DEV  v4.0.18 /home/bszuc/booklore/booklore-ui

 ✓  booklore  src/app/features/magic-shelf/service/magic-shelf-utils.spec.ts (46 tests) 24ms
 ✓  booklore  src/app/app.spec.ts (1 test) 5ms
 ✓  booklore  src/app/features/book/components/book-searcher/book-searcher.component.spec.ts (11 tests) 64ms
 ✓  booklore  src/app/core/security/auth-initializer.spec.ts (2 tests) 39ms
 ✓  booklore  src/app/features/magic-shelf/service/book-rule-evaluator-metadata-presence.spec.ts (85 tests) 247ms
 ✓  booklore  src/app/features/magic-shelf/service/book-rule-evaluator.service.spec.ts (199 tests) 470ms
 ✓  booklore  src/app/app.component.spec.ts (7 tests) 374ms
 ✓  booklore  src/app/features/magic-shelf/component/magic-shelf-component.spec.ts (73 tests) 1279ms

 Test Files  8 passed (8)
      Tests  424 passed (424)
   Start at  14:39:35
   Duration  5.23s (transform 2.58s, setup 2.77s, import 7.66s, tests 2.50s, environment 5.54s)

JUNIT report written to /home/bszuc/booklore/booklore-ui/test-results/vitest-results.xml
 PASS  Waiting for file changes...
       press h to show help, press q to quit

```

</details>

## 📸 Screen Recording / Screenshots (MANDATORY)
image
image

> Every PR must include a **screen recording or screenshots** showing the change working end-to-end in a running local instance (both backend and frontend). This means you must have actually built, run, and tested the code yourself. PRs without visual proof will be closed without review.

---

## ✅ Pre-Submission Checklist

> **All boxes must be checked before requesting review.** Incomplete PRs will be closed without review. No exceptions.

- [X] This PR is linked to an approved issue
- [X] Code follows project style guidelines and conventions
- [X] Branch is up to date with `develop` (merge conflicts resolved)
- [X] I ran the full stack locally (backend + frontend + database) and verified the change works
- [X] Automated tests added or updated to cover changes (backend **and** frontend)
- [X] All tests pass locally and output is pasted above
- [X] Screen recording or screenshots are attached above proving the change works
- [X] PR is a single focused change (one bug fix OR one feature, not multiple unrelated changes)
- [X] PR is reasonably scoped (PRs over 1000+ changed lines will be closed, split into smaller PRs)
- [X] No unsolicited refactors, cleanups, or "improvements" are bundled in
- [ ] Flyway migration versioning is correct _(if schema was modified)_
- [ ] Documentation PR submitted to [booklore-docs](https://github.com/booklore-app/booklore-docs) _(if user-facing changes)_

### 🤖 AI-Assisted Contributions

> **If any part of this PR was generated or assisted by AI tools (Copilot, Claude, ChatGPT, etc.), all items below are mandatory.** You are fully responsible for every line you submit. "The AI wrote it" is not an excuse, and AI-generated PRs that clearly haven't been reviewed are the #1 reason PRs get closed.

- [ ] I have read and understand every line of this PR and can explain any part of it during review
- [ ] I personally ran the code and verified it works (not just trusted the AI's output)
- [ ] PR is scoped to a single logical change, not a dump of everything the AI suggested
- [ ] Tests validate actual behavior, not just coverage (AI-generated tests often assert nothing meaningful)
- [ ] No dead code, placeholder comments, `TODO`s, or unused scaffolding left behind by AI
- [ ] I did not submit refactors, style changes, or "improvements" the AI suggested beyond the scope of the issue

---

### CODE REVIEW

**Comment by alexpicode** [comment]
Will the button option also work if the user presses Enter? If not, perhaps the trigger mode configuration should be set to multi-select.

**Comment by balazs-szucs** [comment]
Yeah, I was also thinking about that. Right now, no. If "button-mode" selected then the enter search does not work. I'm not entirely convinced this is "best-way", so feel free to let me if you'd expect "enter" to work when on "button" mode?

File: booklore-ui/src/i18n/de/settings-view.json [hunk 1]
```diff
@@ -121,5 +121,16 @@
...
+    "sectionDesc": "Configure how search is triggered in the book browser and header search bar.",
+    "searchTrigger": "Search Trigger Mode",
+    "searchTriggerDesc": "'Instant' searches as you type with a debounce delay. 'Search on Enter' waits until you press Enter. 'Search button' adds a button to manually trigger the search.",
+    "modeInstant": "Instant (debounced)",
+    "modeEnter": "Search on Enter",
+    "modeButton": "Search button",
+    "prefsUpdated": "Preferences Updated",
+    "prefsUpdatedDetail": "Your search trigger preference has been saved."
   }
 }
```
Comment (acx10) [inline booklore-ui/src/i18n/de/settings-view.json]: Heads up, the new `search` keys in all the non-en/es language files have English strings as values. These should be empty strings (`""`) so Weblate picks them up for translation. Pushing English here will overwrite whatever translators contribute on the next sync.

File: booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.ts [hunk 2]
```diff
@@ -77,6 +83,18 @@ export class BookSearcherComponent implements OnInit, OnDestroy {
...
+    }
+  }
+
+  onSearchKeydown(event: KeyboardEvent): void {
+    if (event.key === 'Enter' && this.searchMode === 'enter') {
+      this.triggerSearch();
+    }
+  }
+
+  triggerSearch(): void {
```
Comment (acx10) [inline booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.ts:109]: This 500ms debounce in `HeaderFilter` fires even when the user is in enter or button mode. In instant mode it makes sense because switchMap cancels the previous inner observable on each keystroke. But in enter/button mode the user has already committed to searching, so the 500ms delay just feels laggy. Might be worth skipping the debounce when the trigger is explicit.

File: booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            set: vi.fn(),
+          },
+        },
+      ],
+    });
+
+    searchPrefService = TestBed.inject(SearchPreferenceService);
+  });
+
+  function createComponentLogic() {
```
Comment (acx10) [inline booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.spec.ts]: These tests rebuild the component logic in a helper rather than testing the actual `BookSearcherComponent`. If someone changes the real component but not this helper, the tests still pass. I get the DI chain is annoying to set up, but mocking the services and testing the real component would catch more issues.

File: booklore-ui/src/i18n/de/settings-view.json (hunk 1 above)
Comment (balazs-szucs) [reply booklore-ui/src/i18n/de/settings-view.json]: oh. I thought it'd pick it up as untranslated. Sorry, I'm not really familiar with weblate. I'll update that!

File: booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.ts (hunk 2 above)
Comment (balazs-szucs) [reply booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.ts:109]: Makes sense.

File: booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.spec.ts (hunk 3 above)
Comment (balazs-szucs) [reply booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.spec.ts]: I felt like that most error prone part, but nothing against doing the whole thing.

**Comment by alexpicode** [comment]
> Yeah, I was also thinking about that. Right now, no. If "button-mode" selected then the enter search does not work. I'm not entirely convinced this is "best-way", so feel free to let me if you'd expect "enter" to work when on "button" mode?

The standard behavior is usually for 'Enter' to trigger the button's action. I think this would be much more intuitive, especially for mobile users where the keyboard's 'Go/Search' key typically maps to 'Enter'.

**Comment by balazs-szucs** [comment]
Makes sense, I'll update this. THanks for your input!

File: booklore-ui/src/i18n/de/settings-view.json (hunk 1 above)
Comment (balazs-szucs) [reply booklore-ui/src/i18n/de/settings-view.json]: Thanks I addressed these in: [126ec14](https://github.com/booklore-app/booklore/pull/3020/commits/126ec14976c02d84ac743bea768238e8dda2afd8)

Quick question, though, why was (or is?) JSON file formatting inconsistent? For me the script I wrote auto-fixed it, but should revert that or, no?

File: booklore-ui/src/i18n/nl/settings-view.json [hunk 4]
```diff
@@ -1,125 +1,135 @@
 {
```
Comment (acx10) [inline booklore-ui/src/i18n/nl/settings-view.json:1]: This file (and sk, sl, de, it, pl, and all the other non-en/es language files) looks like it got reformatted. The project uses Weblate for community translations, so touching these files risks merge conflicts when Weblate syncs back. Could you revert the formatting changes and only add the new `"search"` block with empty `""` values for non-en/es languages?

File: booklore-ui/src/app/shared/service/search-preference.service.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import {inject, Injectable} from '@angular/core';
+import {BehaviorSubject, Observable} from 'rxjs';
+import {LocalStorageService} from './local-storage.service';
+
+export type SearchTriggerMode = 'instant' | 'button';
```
Comment (acx10) [inline booklore-ui/src/app/shared/service/search-preference.service.ts:4]: The PR title mentions three modes (button, enter, instant) but the type only defines `'instant' | 'button'`. Looks like Enter works as a secondary trigger inside button mode, but there's no standalone enter-only mode. Was that intentional or did it get dropped? Might be worth updating the PR title to match what's actually implemented, or adding the third mode if it was planned.

File: booklore-ui/src/app/features/book/components/book-browser/book-browser.component.ts [hunk 6]
```diff
@@ -766,12 +772,24 @@ export class BookBrowserComponent implements OnInit, AfterViewInit, OnDestroy {
...
+      this.searchTerm$.next(term);
+    }
+  }
+
+  triggerSearch(): void {
+    this.searchTerm$.next(this.bookTitle);
+  }
+
+  onSearchKeydown(event: KeyboardEvent): void {
+    if (event.key === 'Enter' && this.searchMode === 'button') {
```
Comment (acx10) [inline booklore-ui/src/app/features/book/components/book-browser/book-browser.component.ts]: This only triggers on Enter when mode is `'button'`. Might be nice to also let Enter skip the debounce and trigger an immediate search in `'instant'` mode too. Users generally expect Enter to "go" regardless of mode.

File: booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.html [hunk 7]
```diff
@@ -1,74 +1,63 @@
 <ng-container *transloco="let t; prefix: 'book.searcher'">
```
Comment (acx10) [inline booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.html:1]: Looks like the whole template got reformatted alongside the functional changes. Would it be possible to limit the diff to just the lines that actually changed? Makes it much easier to review and reduces the chance of accidental regressions.

File: booklore-ui/src/app/shared/service/search-preference.service.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+const VALID_MODES: SearchTriggerMode[] = ['instant', 'button'];
+const STORAGE_KEY = 'searchTriggerMode';
+
+@Injectable({ providedIn: 'root' })
+export class SearchPreferenceService {
+
+  private readonly localStorageService = inject(LocalStorageService);
+  private readonly modeSubject: BehaviorSubject<SearchTriggerMode>;
+
+  readonly mode$: Observable<SearchTriggerMode>;
```
Comment (acx10) [inline booklore-ui/src/app/shared/service/search-preference.service.ts]: `mode$` doesn't seem to have any subscribers right now. All consumers use the sync `mode` getter. If there's no planned use for the Observable, the service could be simplified to just a getter/setter with localStorage, no need for the BehaviorSubject.

File: booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.scss [hunk 9]
```diff
@@ -32,24 +32,34 @@
 
 .search-input {
   width: 100%;
-  padding-right: 2.5rem;
+  padding-right: 4.5rem; /* space for multiple action buttons */
```
Comment (acx10) [inline booklore-ui/src/app/features/book/components/book-searcher/book-searcher.component.scss:35]: Nit: the `/* space for multiple action buttons */` comment is pretty self-explanatory from the property name and value. Could probably drop it.

File: booklore-ui/src/app/features/settings/view-preferences-parent/search-preferences/search-preferences.component.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+import {FormsModule} from '@angular/forms';
+import {MessageService} from 'primeng/api';
+import {TranslocoDirective, TranslocoService} from '@jsverse/transloco';
+import {SearchPreferenceService, SearchTriggerMode} from '../../../../shared/service/search-preference.service';
+
+@Component({
+  selector: 'app-search-preferences',
+  standalone: true,
+  imports: [
+    FormsModule,
```
Comment (acx10) [inline booklore-ui/src/app/features/settings/view-preferences-parent/search-preferences/search-preferences.component.ts]: `FormsModule` is imported here but the template only uses `(click)` handlers, no `ngModel` or reactive forms. Safe to remove.

File: booklore-ui/src/i18n/nl/settings-view.json (hunk 4 above)
Comment (balazs-szucs) [reply booklore-ui/src/i18n/nl/settings-view.json:1]: But... They were not consistent in the first place...

File: booklore-ui/src/app/shared/service/search-preference.service.ts (hunk 5 above)
Comment (balazs-szucs) [reply booklore-ui/src/app/shared/service/search-preference.service.ts:4]: As per conversion above, I merged the button and enter into one.

File: booklore-ui/src/i18n/nl/settings-view.json (hunk 4 above)
Comment (balazs-szucs) [reply booklore-ui/src/i18n/nl/settings-view.json:1]: These files generally should not be mass edited by hand. I feel like there should CI, that takes care of the formatting and we should just put string in en json.

But yeah just double checked, there were merely formatting inconsistencies and I think the "reformat code" option in Intellij reformatted to the common JSON formatting. They should not be reverted, they make the files consistent.

Comment (balazs-szucs) [reply booklore-ui/src/i18n/nl/settings-view.json:1]: Ok, I went ahead and manually edited each JSON file... Then I committed and realized that this is 100% Intellij doing it. So I had to do it manually again, and commit/push the changes manually for them to stick.

Just FYI, they have inconsistent indenting; IDE would have fixed that. ID, NL, SK, SL, and UK have different indents than the rest. (And guessing by the fact that it's being corrected by the IDE, I'm assuming they were originally created and pushed manually.)

Though, while we're on the topic, we should probably also add an actual Java formatter. This manual stuff has been causing me a huge amount of conflicts for no reason. Running a formatter during the gradlew build would be a good solution.

The OSIV PR had a dozen or so formatting conflicts. No actual code conflicts, just the formatting of imports.

Comment (balazs-szucs) [reply booklore-ui/src/i18n/nl/settings-view.json:1]: So,  #3113 I can see that Weblate also auto-formats the JSONs. So, I think they are safe to edit the way I did, but now if you merge it before this, the PR will get conflicts.

**Comment by alexpicode** [comment]
Hi, why was this PR closed? It's not going to be implemented?

**Comment by balazs-szucs** [comment]
Hi, acx merged another thing #3093, that adds debounce, from my experience this means that this PR is not gonna be merged so not reason to maintain it.

Sorry!
